frontend/public/components/factory/list.tsx

```diff
@@ -230,25 +231,19 @@ export const WorkloadListHeader = props => <ListHeader>
   <ColHead {...props} className="col-lg-3 hidden-md hidden-sm hidden-xs" sortField="spec.selector">Pod Selector</ColHead>
 </ListHeader>;
 
-export class Rows extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.measurementCache = new CellMeasurerCache({
-      fixedWidth: true,
-      minHeight: 50,
-      keyMapper: rowIndex => _.get(this.props.data[rowIndex], 'metadata.uid', rowIndex),
-    });
-
-    this.rowRenderer = this._rowRenderer.bind(this);
-  }
+export const Rows: React.SFC<RowsProps> = (props) => {
+  const measurementCache = new CellMeasurerCache({
+    fixedWidth: true,
+    minHeight: 50,
+    keyMapper: rowIndex => `${_.get(props.data[rowIndex], 'metadata.uid', rowIndex)}-${props.expand ? 'expanded' : 'collapsed'}`,
```

Comment:
This is the core fix of the problem; adding `props.expand` to the row key correctly invalidates the cache and causes a recompute of the height.